File: ui-ngx/src/app/core/http/ai-model.service.ts
// SPDX-FileCopyrightText: Copyright The Thingsboard Authors
// SPDX-License-Identifier: Apache-2.0
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { defaultHttpOptionsFromConfig, RequestConfig } from '@core/http/http-utils';
import { Observable } from 'rxjs';
import { AiModel, AiModelWithUserMsg, CheckConnectivityResult } from '@shared/models/ai-model.models';
import { PageLink } from '@shared/models/page/page-link';
import { PageData } from '@shared/models/page/page-data';

@Injectable({
  providedIn: 'root'
})
export class AiModelService {

  constructor(
    private http: HttpClient
  ) {}

  public saveAiModel(aiModel: AiModel, config?: RequestConfig): Observable<AiModel> {
    return this.http.post<AiModel>('/api/ai/model', aiModel, defaultHttpOptionsFromConfig(config));
  }

  public getAiModels(pageLink: PageLink, config?: RequestConfig): Observable<PageData<AiModel>> {
    return this.http.get<PageData<AiModel>>(`/api/ai/model${pageLink.toQuery()}`, defaultHttpOptionsFromConfig(config));
  }

  public getAiModelById(aiModelId: string, config?: RequestConfig): Observable<AiModel> {
    return this.http.get<AiModel>(`/api/ai/model/${aiModelId}`, defaultHttpOptionsFromConfig(config));
  }

  public deleteAiModel(aiModelId: string, config?: RequestConfig) {
    return this.http.delete(`/api/ai/model/${aiModelId}`, defaultHttpOptionsFromConfig(config));
  }

  public checkConnectivity(aiModelWithUserMsg: AiModelWithUserMsg, config?: RequestConfig): Observable<CheckConnectivityResult> {
    return this.http.post<CheckConnectivityResult>('/api/ai/model/chat', aiModelWithUserMsg, defaultHttpOptionsFromConfig(config));
  }

}

File: common/dao-api/src/main/java/org/thingsboard/server/dao/ai/AiModelService.java
// SPDX-FileCopyrightText: Copyright The Thingsboard Authors
// SPDX-License-Identifier: Apache-2.0
package org.thingsboard.server.dao.ai;

import com.google.common.util.concurrent.FluentFuture;
import org.thingsboard.server.common.data.ai.AiModel;
import org.thingsboard.server.common.data.id.AiModelId;
import org.thingsboard.server.common.data.id.TenantId;
import org.thingsboard.server.common.data.page.PageData;
import org.thingsboard.server.common.data.page.PageLink;
import org.thingsboard.server.dao.entity.EntityDaoService;

import java.util.Optional;

public interface AiModelService extends EntityDaoService {

    AiModel save(AiModel model);

    Optional<AiModel> findAiModelById(TenantId tenantId, AiModelId modelId);

    PageData<AiModel> findAiModelsByTenantId(TenantId tenantId, PageLink pageLink);

    Optional<AiModel> findAiModelByTenantIdAndId(TenantId tenantId, AiModelId modelId);

    FluentFuture<Optional<AiModel>> findAiModelByTenantIdAndIdAsync(TenantId tenantId, AiModelId modelId);

    boolean deleteByTenantIdAndId(TenantId tenantId, AiModelId modelId);

}
